### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by VoTri**
Title
Chapter 7 of "Ruby on Rails Tutorial (Rails 5)"

Background
Training Rails

Technical Changes
Add routes users
Add page sign-up
Test sign-up information

Database Migration
Need to run rails db:migrate:reset.

### CODE REVIEW

File: app/views/shared/_error_messages.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (anonymept) [inline app/views/shared/_error_messages.html.erb]: Please indent this block

File: config/locales/en.yml [hunk 2]
```diff
@@ -48,5 +48,8 @@ en:
     users:
       new:
         sign_up: "Sign up"
-        sign_up_page: "This will be a signup page for new users."
+        create_my_account: "Create my account"
+  controllers:
+    users:
+        welcome_sample_app: "Welcome to the Sample App!"
```
Comment (anonymept) [inline config/locales/en.yml]: use 2 spaces instead of 4

File: db/schema.rb [hunk 3]
```diff
@@ -18,6 +18,7 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.string "password_digest"
+    t.index ["email"], name: "index_users_on_email", unique: true
```
Comment (anonymept) [inline db/schema.rb:21]: [Q] from where you get it ?

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,3 +1,29 @@
...
+      flash[:success] = t("controllers.users.welcome_sample_app")
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  private
+
+    def user_params
```
Comment (anonymept) [inline app/controllers/users_controller.rb]: our style:
- No blank line after `private` (optional)
- functions have same alignment with `private`
Please try `rubycop` if you have time:
https://github.com/rubocop-hq/rubocop

File: db/schema.rb (hunk 3 above)
Comment (VoTri) [reply db/schema.rb:21]: in chapter 6, i doesn't run rails db:migrate. chapter 7, i run rails db:migrate:reset.

**Pull request merged**